processor/simple_supply_tp/handler.py

```diff
@@ -27,11 +39,40 @@ def family_name(self):
 
     @property
     def family_versions(self):
-        return ['0.0']
+        return [addresser.FAMILY_VERSION]
 
     @property
     def namespaces(self):
         return [addresser.NAMESPACE]
 
     def apply(self, transaction, context):
-        raise InternalError('Not implemented')
+        header = transaction.header
+        payload = SimpleSupplyPayload(transaction.payload)
+        state = SimpleSupplyState(context)
+
+        _validate_timestamp(payload.timestamp)
+
+        if payload.action == payload_pb2.SimpleSupplyPayload.CREATE_AGENT:
+            _create_agent(
+                state=state,
+                public_key=header.signer_public_key,
+                payload=payload)
+        else:
+            raise InvalidTransaction('Unhandled action')
+
+
+def _create_agent(state, public_key, payload):
+    if state.get_agent(public_key):
+        raise InvalidTransaction('Account with the public key {} already '
+                                 'exists'.format(public_key))
+    state.set_agent(
+        public_key=public_key,
+        name=payload.data.name,
+        timestamp=payload.timestamp)
+
+
+def _validate_timestamp(timestamp):
```

Comment:
IIRC SupplyChain initially had something like this, but it was taken out because there was something bad about it. I would double check with someone who knows the details and make sure this is needed and that it is correct.

Reply:
@peterschwarz @delventhalz Any thoughts about this?

Reply:
Decided to add a tolerance of 5 mins to mitigate out of sync system clocks